File: android/app/src/main/java/com/example/pr_penjualanhandphone_ardisaputratip212/Petugas.java
package com.example.pr_penjualanhandphone_ardisaputratip212;

public class Petugas extends koneksi{
    server server = new server();
    String conn = server.conn();
    String URL = conn + "/PR_PenjualanHandPhone_ArdiSaputraTIP-21.2/server/petugas.php";
    String url = "";
    String response = "";
    public void index(ResponseCallback callback){
        try {
            url = URL + "?operasi=tampil";
            runAsync(url, callback);
        } catch (Exception e){
            e.printStackTrace();
            callback.onFailure(e);
        }
    }
    public String store(String nama, String alamat, String notelp, String jabatan){
        try {
            url = URL + "?operasi=tambah&nama=" + nama + "&alamat=" + alamat + "&notelp="  + notelp + "&jabatan=" + jabatan;
            response = run(url);
        }catch (Exception e){
            e.printStackTrace();
            e.getMessage();
        }
        return response;
    }
    public String getData(int id){
        try{
            url = URL + "?operasi=getdata&id=" + id;
            response = run(url);
        }catch (Exception e){
            e.printStackTrace();
            e.getMessage();
        }
        return response;
    }
    public String update(String id, String nama, String alamat, String notelp, String jabatan){
        try{
            url = URL + "?operasi=update&nama=" + nama + "&alamat=" + alamat + "&notelp="  + notelp + "&jabatan=" + jabatan + "&id=" + id;
            response = run(url);
        } catch (Exception e){
            e.printStackTrace();
            e.getMessage();
        }
        return response;
    }
    public String destroy(int id){
        try {
            url = URL + "?operasi=delete&id=" + id;
            response = run(url);
        } catch (Exception e){
            e.printStackTrace();
            e.getMessage();
        }
        return response;
    }
}

File: server/petugas.php
<?php
//include 'database.php';
require_once 'DatabaseConnection.php';

$database = new Database();

$tablename = 'petugas';
$operasi = $_GET['operasi'];

if (isset($_GET['nama'], $_GET['alamat'], $_GET['notelp'], $_GET['jabatan'])) {
    $data = [
        'nama_petugas' => $_GET['nama'],
        'alamat' => $_GET['alamat'],
        'no_telp' => $_GET['notelp'],
        'jabatan' => $_GET['jabatan']
    ];
}

if(isset($_GET['id'])){
    $condition = [
        'id_petugas' => $_GET['id']
    ];
}
//var_dump($data);

switch ($operasi) {
    case 'tampil':
        $database->index($tablename);
        //echo json_encode($datakirim);
    break;
    
    case 'tambah':
        //var_dump($data);
        $database->store($data, $tablename);
    break;

    case 'getdata':
        $database->detailData($condition, $tablename);
    break;

    case 'update':
        $database->update($data, $tablename, $condition);
    break;

    case 'delete':
        $database->destroy($condition, $tablename);
    break;
    
    default: break;
}